core/src/main/scala/org/apache/spark/resource/ResourceProfileManager.scala

```diff
@@ -57,6 +57,7 @@ private[spark] class ResourceProfileManager(sparkConf: SparkConf,
   private val isStandaloneOrLocalCluster = master.isDefined && (
       master.get.startsWith("spark://") || master.get.startsWith("local-cluster")
     )
+  private val supportTaskResourceProfile = isStandaloneOrLocalCluster || isYarn
```

Comment:
We should either inline it, or also check whether DRA is disabled - else the variable is inaccurate.

```suggestion
  private val supportTaskResourceProfile = isStandaloneOrLocalCluster || (isYarn && !dynamicEnabled)
```

Reply:
Good suggestion. Talked with Tom, seems `inline` is a better choice, avoiding checking DRA twice.